modules/miner/miner.go

```diff
@@ -165,7 +169,17 @@ func New(cs modules.ConsensusSet, tpool modules.TransactionPool, w modules.Walle
 		headerMem:  make([]types.BlockHeader, HeaderMemory),
 
 		fullSets:  make(map[modules.TransactionSetID][]int),
-		splitSets: make(map[int]*splitSet),
+		splitSets: make(map[splitSetID]*splitSet),
+		blockMapHeap: &mapHeap{
+			selectID: make(map[splitSetID]*mapElement),
+			data:     make([]*mapElement, 0),
+			minHeap:  true,
+		},
+		overflowMapHeap: &mapHeap{
+			selectID: make(map[splitSetID]*mapElement),
+			data:     make([]*mapElement, 0),
+			minHeap:  false,
+		},
```

Comment:
maybe have a `newMapHeap` constructor? Also, it's not common to call `make` with `0` length. You can use `nil` instead or omit the field entirely.